src/ds5/ds5-thermal-handler.cpp

```diff
@@ -0,0 +1,188 @@
+        if (state != _control_on)
+        {
+            _tl_activation->set(state);
+            _control_on = state;
+            update_mode();
+        }
+    }
+
+    float ds5_thermal_handler::query()
+    {
+        auto ctrl_state = static_cast<bool>(_tl_activation->query());
+        LOG_DEBUG(__FUNCTION__ << " _control_on = " << _control_on << " ctrl_state = " << ctrl_state);
+        if (_control_on != ctrl_state)
+        {
+            _control_on = ctrl_state;
+            update_mode();
+        }
+        return ctrl_state;
+    }
+
+
+    void ds5_thermal_handler::update_mode(bool on_streaming)
+    {
+        if (_streaming_on && _control_on)
+        {
+            start();
+            return;
+        }
+        // Deactivate when toggling off streaming or control
+        if ((!_streaming_on && _control_on && on_streaming) ||
+            (_streaming_on && !_control_on && !on_streaming))
+        {
+            stop();
+            return;
+        }
+
+        LOG_ERROR("Thermal compensation update_mode was not changed: on_streaming =" << on_streaming
+                    << ", _streaming_on=" << _streaming_on << ", _control_on=" << _control_on);
+    }
+
+    void ds5_thermal_handler::polling(dispatcher::cancellable_timer cancellable_timer)
+    {
+        if (cancellable_timer.try_sleep(_poll_intervals_ms))
+        {
+            try
+            {
+                auto ts = (uint64_t)std::chrono::high_resolution_clock::now().time_since_epoch().count();
+                if (auto sp = _dpt_sensor.lock())
+                {
+                    auto cur_temp = sp->get_option(RS2_OPTION_ASIC_TEMPERATURE).query();
+
+                    if (fabs(_temp_base - cur_temp) >= _thermal_threshold_deg)
+                    {
+                        if (auto recalib_p = _recalib_sensor.lock())
+                        {
+                            LOG_INFO("Thermal calibration adjustment is triggered on change from "
+                                << std::dec << _temp_base << " to " << cur_temp << " deg (C)");
+                            recalib_p->reset_calibration();
+
+                            notify_of_calibration_change(RS2_CALIBRATION_SUCCESSFUL);
+
+                            _temp_base = cur_temp;
+                        }
+                    }
+                }
+                else
+                {
+                    LOG_ERROR("Thermal Compensation: Depth sensor is not present");
+                }
+            }
+            catch (const std::exception& ex)
+            {
+                LOG_ERROR("Error during thermal compensation handling: " << ex.what());
+            }
+            catch (...)
+            {
+                LOG_ERROR("Unresolved error during Thermal Compensation handling!");
+            }
+        }
+        else
+        {
+            LOG_DEBUG("Thermal Compensationis being shut-down");
+        }
```

Comment:
typo - add a space between "Compensation" and "is"

Reply:
good catch 👍